Add annotated slice dropdown

cleaaum opened the pull request:
This PR adds a dropdown with the currently annotated slices. When a slice has annotations on it, this dropdown gets populated so the user can easily find which slices were annotated.

Implementation notes:
- When a user selects a slice to jump to, the dropdown value clears after it has been selected. This is to avoid confusion when a user goes to a different slice, there is no mismatch between the current slice and the slice in the dropdown. 
- This dropdown gets updated every time an annotation is drawn, when a class is deleted, when all annotations are deleted on a slice (and when a specific annotation is deleted via the eraser).
- May need to rework/rephrase the UI


<img width="411" alt="Screen Shot 2023-09-25 at 11 31 21 AM" src="https://github.com/mlexchange/mlex_highres_segmentation/assets/56569375/9b87284a-4e29-459f-89bd-fbddd33d34d4">

Closes #122

danton267 commented:
Minor tweak suggestion: when the user selects a slice that is currently active do not "redraw it", add raise PreventUpdate to `render_image`

otherwise, it looks good! (works even with loading!) 🥇

danton267 reviewed (dismissed):
Minor tweak suggestion: when the user selects a slice that is currently active do not "redraw it", add raise PreventUpdate to render_image

Minor tweak 2: add a notification "annotated slice x has just been selected"

otherwise, it looks good! (works even with loading!) 🥇

hannahker reviewed (approved):
Great! Agree with @danton267's point on raising the `PreventUpdate` if you're already on the slice that you selected. It could be nice to also add a tooltip around the control (or just the label) that says something to the effect of "Jump to your annotated slices"

hannahker commented on components/control_bar.py:
```diff
@@ -156,6 +156,15 @@ def layout():
                                 ),
+                                dmc.Space(h=25),
+                                _control_item(
+                                    "Annotated Slices",
```
Very nit-picky but I'd leave "slices" lowercase here.

cleaaum commented:
I've addressed all the comments! 
- new notification added
- tooltip added
- label capitalization fixed
- prevent update added when picking slice on current slice

I've also refactored the notifications a bit. I created a function that generates a notification, you just pass the title, color and icon. I've also realized half the notification items are stored in a constants dict so I added all of the messages and icons to that dict for consistency.

hannahker commented on components/control_bar.py:
```diff
@@ -24,12 +24,15 @@ def _control_item(title, title_id, item):
-                size="sm",
-                style={"width": "100px", "margin": "auto", "paddingRight": "5px"},
-                align="right",
+            _tooltip(
```
This one is adding that tooltip to all the instances of the `_control_item()` 🌀

cleaaum replied:
oh dang good catch I didn't realize that!! Ill fix that:)

cleaaum replied:
Fixed!

hannahker commented:
@cleaaum LGTM! 💃
